src/ui/public/index_patterns/_field.js

```diff
@@ -40,10 +40,10 @@ export default function FieldObjectProvider(Private, shortDotsFilter, $rootScope
 
     const indexed = !!spec.indexed;
     const scripted = !!spec.scripted;
-    const sortable = spec.name === '_score' || ((indexed || scripted) && type.sortable);
-    const filterable = spec.name === '_id' || scripted || (indexed && type.filterable);
     const searchable = !!spec.searchable || scripted;
     const aggregatable = !!spec.aggregatable || scripted;
+    const sortable = spec.name === '_score' || ((indexed || scripted || aggregatable) && type.sortable);
+    const filterable = spec.name === '_id' || scripted || ((indexed || searchable) && type.filterable);
```

Comment:
This is probably me not knowing enough about the various `sortable` and `filterable` attributes, but the asymmetry to the previous line has caught my eye. So a `scripted` field's type has to be `sortable` in order for the field to be `sortable`, but its type does not have to be `filterable` for the field to be `filterable`?